### ISSUE TITLE
[FR] Ability to resize the arrow heads in e.g. quiver plots

### ISSUE THREAD

**Issue by TorkelE** [JuliaPlots/Plots.jl#4881]
I have been trying to figure out how to do this (trying tons of various options), but currently does not seem to be possible in Plots.

It would be useful to be able to resize the arrowheads in e.g. quiver plots. E.g. for
```julia
using Plots
xs = vcat([fill(i,10) for i in 1:10]...)
ys = vcat(fill(1:10,10)...)
dxs = (0.8 .- rand(100)) ./2
dys = (0.6 .- rand(100)) ./2
quiver(xs, ys; quiver=(dxs,dys))
```
![image](https://github.com/JuliaPlots/Plots.jl/assets/18099310/92de8232-4493-4bb7-b319-d63e40c0c189)

Especially when the arrow lines are small, the arrowheads looks really weird and ugly.

**Comment by j-adel** [comment]
I really suggest this [`arrow!()` function](https://discourse.julialang.org/t/plots-jl-arrows-style-in-quiver/13659/6). This issue has been there for over 7 years. I think it's time...

**Comment by BeastyBlacksmith** [comment]
You are welcome to open a PR 🙂

**Comment by j-adel** [comment]
> You are welcome to open a PR 🙂

It seems to me that `function quiver_using_arrows(plotattributes::AKW)` is already designed to have scaling arrow heads. I'm not sure if there's something preventing its use or if there's a way for the user to select it.

**Comment by BeastyBlacksmith** [comment]
True, but that is only used, if the backend supports the `:arrow` attribute
```julia
@recipe function f(::Type{Val{:quiver}}, x, y, z)  # COV_EXCL_LINE
    @nospecialize
    if :arrow in supported_attrs()
        quiver_using_arrows(plotattributes)
    else
        quiver_using_hack(plotattributes)
    end
    ()
end
```

which GR currently does not: 
https://github.com/JuliaPlots/Plots.jl/blob/da79316e5bd768e417f65f344b7d749e41bbb0be/PlotsBase/ext/GRExt.jl#L25

That is why it uses the other implementation

**Comment by j-adel** [comment]
The problem isn't in `quiver_using_hack` (which is actually not triggered in GR. In fact, `hack` has scaling arrow heads:
![image](https://github.com/user-attachments/assets/a788ea87-1234-4cfe-ac08-280181ad393a)
It's the `quiver_using_arrows()` that's problematic. But I really don't understand the library structure to see where the line vectors are being computed.

**Comment by j-adel** [comment]
I see. But it seems like the function utilizes `GR.drawarrow()` which doesn't take more than the coordinates, the direction, and a preset style.

https://github.com/JuliaPlots/Plots.jl/blob/da79316e5bd768e417f65f344b7d749e41bbb0be/PlotsBase/ext/GRExt.jl#L391:392

From the GR backend https://gr-framework.org/_modules/gr.html#setarrowstyle, it seems like you can't set the arrow head to automatically scale. But there is also a setarrowsize(size). Maybe that's worth trying? I'm sorry but I'm busy these couple of weeks so I can't dive into GR backend at this point.
